### PR TITLE
Зырина -- Лабораторная работа #1 

### PR DESCRIPTION
**Pull request by Selllavi**
I was waiting for 0 lab merge, because i have Test file in this merge too, because i have merged my 0 lab in master in my origin repo. (My mistake, I should not have done this)
But it is too risky wait for 0 lab merge now. 
Please, do not pay attention on Test file.

### CODE REVIEW

**Comment by Selllavi** [comment]
@ViolettaShirokikh, @Yaroslav-Lahtachev please, review these changes.

**Comment by ViolettaShirokikh** [comment]
What sorting algorithm did you use?

File: code/lab1-tdd-sorting/src/main/java/ru/unn/agile/sorting/impl/NumberSorting.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        return numbers;
+    }
+
+    private boolean compareTwoNumbers(final int number1, final int number2) {
+        if (getDirection().equals(Direction.ASC)) {
+            return getExpression().compare(number1, number2);
+        } else {
+            return !getExpression().compare(number1, number2);
+        }
+    }
```
Comment (ViolettaShirokikh) [inline code/lab1-tdd-sorting/src/main/java/ru/unn/agile/sorting/impl/NumberSorting.java]: Duplicate code "getExpression().compare(number1, number2)". Maybe save the result of this method in separate variable?

File: code/lab1-tdd-sorting/src/main/java/ru/unn/agile/sorting/impl/NumberSorting.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    public NumberSorting(final Direction direction, final Expression<Integer> expression) {
+        super(direction, expression);
+    }
+
+    @Override
+    public Integer[] sort(final Integer[] numbers) {
+        for (int i = 1; i < numbers.length; i++) {
+            int current = numbers[i];
+            int j = i - 1;
+            while (j >= 0 && compareTwoNumbers(current, numbers[j])) {
```
Comment (ViolettaShirokikh) [inline code/lab1-tdd-sorting/src/main/java/ru/unn/agile/sorting/impl/NumberSorting.java]: Shouldn't "j >=0" be written in brackets?

File: code/lab1-tdd-sorting/src/main/java/ru/unn/agile/sorting/impl/NumberSorting.java (hunk 1 above)
Comment (Selllavi) [reply code/lab1-tdd-sorting/src/main/java/ru/unn/agile/sorting/impl/NumberSorting.java]: yeah, you are right

**Comment by Selllavi** [comment]
> What sorting algorithm did you use?

Insertion Sort https://en.wikipedia.org/wiki/Insertion_sort

File: code/lab1-tdd-sorting/src/main/java/ru/unn/agile/sorting/impl/NumberSorting.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    public NumberSorting(final Direction direction, final Expression<Integer> expression) {
+        super(direction, expression);
+    }
+
+    @Override
+    public Integer[] sort(final Integer[] numbers) {
+        for (int i = 1; i < numbers.length; i++) {
+            int current = numbers[i];
```
Comment (Yaroslav-Lahtachev) [inline code/lab1-tdd-sorting/src/main/java/ru/unn/agile/sorting/impl/NumberSorting.java]: Maybe change int to Integer for uniformity and convertation avoid.

**Review by Yaroslav-Lahtachev — COMMENTED** [review]
And add toString, hashCode, equals methods

**Comment by Selllavi** [comment]
@ViolettaShirokikh @Yaroslav-Lahtachev  please, see again

**Comment by ViolettaShirokikh** [comment]
:+1:

**Comment by Yaroslav-Lahtachev** [comment]
good :+1:

File: code/lab1-tdd-sorting/src/test/java/ru/unn/agile/Sorting/NumberSortingTest.java [hunk 4]
```diff
@@ -0,0 +1,197 @@
+package ru.unn.agile.Sorting;
```
Comment (might52) [inline code/lab1-tdd-sorting/src/test/java/ru/unn/agile/Sorting/NumberSortingTest.java]: Please fix the package name. It should be low case.

**Pull request merged**